Deploy demos with Jenkins

pmac opened the pull request:
Deploy demos from the Jenkinsfile

pmac commented:
This works. I deployed `demo__pmac-jenkins` with it. I need to do some more cleanup before merge however, and I wanted to check with @jgmize and @glogiotatidis before going forward to make sure it looks okay. It's WAY faster though :) I've decided to skip any tests of the demos at this point. They'll have to be fully tested in PR before being merged anyway, but we can discuss that as well.

glogiotatidis commented on docker/jenkins/demo_deploy.sh:
```diff
@@ -0,0 +1,24 @@
+set -e
+
+GIT_COMMIT=$(git rev-parse HEAD)
+DEIS_APP_NAME="bedrock-demo-${BRANCH_NAME#demo__}"
```
👍 Didn't know about `${var#Pattern}`

glogiotatidis commented:
lgtm, ping me again for another pass when ready. 🎄

pmac commented:
I believe this is now ready for a full review and merge. I cleaned up the circle scripts and have added demo Sentry DSN secret support. Thanks again for the review @glogiotatidis and @jgmize. I've been testing this in the `demo__pmac-jenkins` branch, which you can [see in Jenkins](https://ci.us-west.moz.works/view/Bedrock/job/bedrock_multibranch_pipeline/job/demo__pmac-jenkins/).

jgmize commented on docker/jenkins/demo_deploy.sh:
```diff
@@ -0,0 +1,26 @@
+docker tag "bedrock_dev_final:${GIT_COMMIT}" "localhost:5001/$DOCKER_IMAGE_TAG"
+
+echo "Pushing ${DOCKER_IMAGE_TAG} to docker repository"
+docker push "localhost:5001/${DOCKER_IMAGE_TAG}"
```
NIT: "localhost:5001" here and a few lines above should be a variable IMO, perhaps REGISTRY?

pmac replied:
Good call. I used `$PRIVATE_REGISTRY` to be consistent with existing scripts and put the definition in the Jenkinsfile.

pmac commented:
Updated and the tests and demo deploy are running again. I'll let Giorgos weigh in and then we can hopefully land this first thing 2017. 

Happy new year all 🎆 🎉

pmac commented on Jenkinsfile:
```diff
@@ -9,8 +10,24 @@ else if ( branch == 'prod') {
     build 'bedrock_base_image'
 }
 else if ( branch ==~ /^demo__[a-z_-]+$/ ) {
```
oh snap! good catch. definitely meant to include that.

pmac commented:
More updates:

* Added IRC notification support (Thanks @glogiotatidis and @davehunt)
* Fixed tag and release docs
* Fixed deploying demos with numbers in the branch

I think we're really for real ready this time 🤞

pmac commented:
NOTE: I've also turned on the "only build pull-requests" setting in circleci. So we shouldn't be wasting time and notifications building every branch now.
